pkg/ingester/otlp/ingest_handler.go

```diff
@@ -65,41 +67,46 @@ func (h *ingestHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 
 // TODO(@petethepig): split http and grpc
 func (h *ingestHandler) Export(ctx context.Context, er *pprofileotlp.ExportProfilesServiceRequest) (*pprofileotlp.ExportProfilesServiceResponse, error) {
-
-	// TODO(@petethepig): make it tenant-aware
-	ctx = tenant.InjectTenantID(ctx, tenant.DefaultTenantID)
-
-	h.log.Log("msg", "Export called")
+	// Extracts user ID from the request metadata and returns and injects the user ID in the context
+	_, ctx, err := user.ExtractFromGRPCRequest(ctx)
+	if err != nil {
+		level.Error(h.log).Log("msg", "failed to extract tenant ID from GRPC request", "err", err)
+		return &pprofileotlp.ExportProfilesServiceResponse{}, fmt.Errorf("failed to extract tenant ID from GRPC request: %w", err)
+	}
 
 	rps := er.ResourceProfiles
 	for i := 0; i < len(rps); i++ {
 		rp := rps[i]
 
-		labelsDst := []*typesv1.LabelPair{}
-		// TODO(@petethepig): make labels work
-		labelsDst = append(labelsDst, &typesv1.LabelPair{
-			Name:  "__name__",
-			Value: "process_cpu",
-		})
-		labelsDst = append(labelsDst, &typesv1.LabelPair{
-			Name:  "service_name",
-			Value: "otlp_test_app4",
-		})
-		labelsDst = append(labelsDst, &typesv1.LabelPair{
-			Name:  "__delta__",
-			Value: "false",
-		})
-		labelsDst = append(labelsDst, &typesv1.LabelPair{
-			Name:  "pyroscope_spy",
-			Value: "unknown",
-		})
+		// Get service name
+		serviceName := getServiceNameFromAttributes(rp.Resource.GetAttributes())
+		fmt.Println("service.name: ", serviceName)
+
+		// Start with default labels
+		labels := getDefaultLabels(serviceName)
+
+		// Track processed attribute keys to avoid duplicates across levels
+		processedKeys := make(map[string]bool)
+
+		// Add resource attributes
+		labels = appendAttributesUnique(labels, rp.Resource.GetAttributes(), processedKeys)
```

Comment:
Right now we're adding all attributes as labels... do we want to set some limit?